File: python/flatten-array/flatten_array.py
def flatten(nested_list):
    return [element for element in flatten_list(nested_list)]


def flatten_list(nested_list):
    for element in nested_list:
        if not isinstance(element, list):
            if element is not None and not isinstance(element, tuple):
                yield element
        else:
            for subelement in flatten_list(element):
                yield subelement

File: go/flatten-array/flatten_array.go
package flatten

func Flatten(nested interface{}) []interface{} {
	result := []interface{}{}
	for _, element := range nested.([]interface{}) {
		if element == nil {
			continue
		}
		if slice, ok := element.([]interface{}); ok {
			result = append(result, Flatten(slice)...)
		} else {
			result = append(result, element)
		}
	}
	return result
}
